src/main/java/com/iexec/worker/executor/TaskNotificationService.java

```diff
@@ -228,21 +233,24 @@ private boolean storeWorkerpoolAuthAndSmsFromExtraIfPresent(TaskNotificationExtr
     }
 
     private TaskNotificationType updateStatusAndGetNextAction(String chainTaskId,
-                                                              ReplicateStatus status) {
+                                                              ReplicateStatus status)
+            throws InterruptedException {
```

Comment:
I would avoid this pollution here and implement the following code directly in the `waitForSessionReady` method
Use SonarLint and follow its recommendation on this one
```
try {
} catch (InterruptedExecption e) {
    Thread.currentThread().interrupt()
}

```

Reply:
I prefer it to be part of the calling methods. Please see https://github.com/iExecBlockchainComputing/iexec-worker/pull/497/commits/33b0892c7336fb67f8741c9eab60f9f496440f08.